### PR TITLE
OUSD vault upgrade script

### PR DESCRIPTION
**Pull request by naddison36**
Upgrades the OUSD Vault to the latest `VaultCore` and `VaultAdmin` contracts.
This includes cached decimals and collateral swaps.

### CODE REVIEW

**Comment by shahthepro** [comment]
Tagging one relevant issue here, so that we don't miss it: #1368

**Review by sparrowDom — APPROVED** [review]
LGTM

**Comment by DanielVF** [comment]
@sparrowDom Did we have an old deploy file that did this? If so, we should check for any differences, and then remove the old file?

File: contracts/deploy/067_ousd_eth_vault_upgrade.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          contract: cVault,
+          signature: "cacheDecimals(address)",
+          args: [assetAddresses.USDC],
+        },
+        {
+          contract: cVault,
+          signature: "cacheDecimals(address)",
+          args: [assetAddresses.USDT],
+        },
+        // TODO cache other decimals
```
[the remark this answers is not in the document]
Comment (naddison36) [reply contracts/deploy/067_ousd_eth_vault_upgrade.js:72]: I wasn't sure if assets like COMP, Aave, CRV and CVX needed to be cached. It doesn't look like the Vault uses these so I'll remove the TODO

**Comment by naddison36** [comment]
> @sparrowDom Did we have an old deploy file that did this? If so, we should check for any differences, and then remove the old file?

@DanielVF It looks like `052_decimal_cache` which doesn't have a block number in the migration file.

#52 also deploys a new `OracleRouter` and `Harvester`. 
It also caches COMP, Aave, CRV and CRV.

I'd suggest we stick to deploy #52 and just close this PR

**Comment by naddison36** [comment]
Setting the swapper config will have to be added to the 52 deploy script. I can do that once collateral swaps is finalised.

**Comment by sparrowDom** [comment]
Yes I agree @naddison36 we can add the Swapper deployment to the 052 script and also the setOracleSlippage on the Vault. 

just FYI You've cached all the required assets on the Vault in this PR. The 052 also caches all the assets and reward tokens on the Oraclerouter. If the reward tokens wouldn't be cached on the OracleRouter the Harvester would fail when it would try to harvest the rewards ( [see this line](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/harvest/Harvester.sol#L64) )

**Comment by naddison36** [comment]
I've changed this PR to update the existing #52 deploy script to include the swap config

**Pull request merged**